pennylane_lightning/src/observables/tests/Test_Observables.cpp

```diff
@@ -31,12 +31,29 @@ constexpr bool BACKEND_FOUND = true;
 
 #include "TestHelpersStateVectors.hpp" // TestStateVectorBackends, StateVectorToName
 
+#define ISAPPROXEQUAL(A, B)                                                    \
+    isApproxEqual(A.getData(), A.getLength(), B.data(), B.size())
```

Comment:
Right now, I introduced a macro in a couple tests to wrap `isApproxEqual` because it does not take in a `Kokkos::View`. There are some options: 

- I could defined a wrapper function in the tests instead.
- Overload `isApproxEqual` and overload it in a LKokkos header.
- Have statevector types define a generic `getDataVector` method that returns a `std::vector` type (with copies if necessary) providing a flexible/robust type for validation. 

@AmintorDusko preferred not to have `getDataVector` to keep SVRaw as light as possible.

Reply:
It was more than that, though. The RawStateVector attaches itself to a pointer for externally stored data.
Returning a std::vector from it would imply allocating more data to create such a vector, and such a method would only be used (so far) for tests. I think we should avoid changing the state vector to accommodate tests (if possible) and solve test problems with tests specific functionality, like a TestHelper.
Adding to that, what I proposed in one of my comments will problem solve this problem.

Reply:
My bad, I didn't meant to sweep your arguments under the rug. I think that this is a good point

> we should avoid changing the state vector to accommodate test

and I agree your solution is ideal (getData => getView), but I thought it is a little dangerous to change before adding the bindings. On further inspection, it seems `getData` is not called anywhere in the bindings. I'll give it a shot.